### PR TITLE
Adds documention for development and events/blog posts

### LINKED ISSUE #51: Add documentation
We should document setting up and running the project, as well as what is necessary to contribute to blog pages or events.

### PR DESCRIPTION
**Pull request by tylersmalley**
Placed everything in the readme for now. Once it gets too big we can split it out into other locations.

Closes #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,135 @@
 # tailscale.dev
 
-## MDX Components
+This site is powered by [Next.js](https://nextjs.org/), [Tailwind CSS](https://tailwindcss.com/) and [ContentLayer](https://www.contentlayer.dev/).
+
+## Local Development
+
+**Quick start**
+
+Running `tools/yarn install` will install all required dependencies. Once complete, use `tools/yarn dev` to start the development server which defaults to port `3000`.
```
Comment (shayne) [inline README.md:9]: `tool` vs `tools`

File: README.md [hunk 2]
```diff
@@ -1,14 +1,135 @@
...
+
+Following the [Tailscale OSS](https://github.com/tailscale/tailscale) repository, we use a `./tool` directory to manage tool dependencies. Versions are pinned via `*.rev` files in the projects' root and installed via `./tool/redo.sh` using the `*.cmd.do` files also in the project's root.
+
+The following tools are available:
+
+- `./tool/node` - [Node](https://nodejs.org/) for future JavaScript tooling
+- `./tool/yarn` - [Yarn](https://yarnpkg.com/) package manager
+- `./tool/redo.sh` - [Redo](https://github.com/apenwarr/redo) build/automation tool (for deps)
+
+If avaiable, direnv will place these tools in your PATH per our `.envrc` config. Installation instructions for direnv are available [here](https://direnv.net/docs/installation.html).
```
Comment (shayne) [inline README.md:21]: Link to direnv project?

File: README.md [hunk 3]
```diff
@@ -1,14 +1,135 @@
...
+## Local Development
+
+**Quick start**
+
+Running `tools/yarn install` will install all required dependencies. Once complete, use `tools/yarn dev` to start the development server which defaults to port `3000`.
+
+**Development tools**
+
+Following the [Tailscale OSS](https://github.com/tailscale/tailscale) repository, we use a `./tool` directory to manage tool dependencies. Versions are pinned via `*.rev` files in the projects' root and installed via `./tool/redo.sh` using the `*.cmd.do` files also in the project's root.
+
```
Comment (shayne) [inline README.md:14]: Mention the flake for nix users and `nix develop` for the environment.

**Review by shayne — CHANGES_REQUESTED** [review]

**Comment by tylersmalley** [comment]
@shayne, I pushed https://github.com/tailscale-dev/tailscale-dev/pull/52/commits/1caadd450a0ba508190a96d13d9b496041579a61 to address your feedback.

**Review by shayne — APPROVED** [review]

**Pull request merged**